src/inspectorscripts.ts

```diff
@@ -77,7 +77,9 @@ def _jupyterlab_variableinspector_getsizeof(x):
     elif __torch and isinstance(x, __torch.Tensor):
         return x.element_size() * x.nelement()
     elif __pd and type(x).__name__ == 'DataFrame':
-        return x.memory_usage().sum()
+        # DO NOT CALL df.memory_usage() as this can be very costly
+        # to the point of crashing the kernel
+        return "?"
```

Comment:
```suggestion
        with pd.option_context("mode.copy_on_write", True):
            return x.memory_usage().sum()
```

Reply:
Nope, still the same issue with this.

Running a profiler on inspecting a 10 rows x 500_000 columns I see this takes more than 6 seconds to run. When inspecting a 10 rows x 50_000_000 columns, laptop goes out of memory and crashes.

Without doing any memory_usage computation, inspection takes less than 10 ms.

Reply:
Thanks for the suggestion though!

I wonder if we could add a small condition on the shape, if the shape is in the 10 thousands we don't compute the memory usage?

Reply:
Ok, what about using a lazy approximation?

```suggestion
        return x.head(1).memory_usage().sum() * len(x)
```

Reply:
That is still quite slow when there are many columns (500_000 here, 6.8 seconds to compute)

![Screenshot from 2024-09-04 14-42-27](https://github.com/user-attachments/assets/7331339d-89bc-47c1-832e-faee571455b2)

Note that this means adding 7 seconds of delay between each cell execution once the variable is defined.

Reply:
Hmm, I guess we could use some kind of cache. I guess it is feasible to write a function giving a rough estimate with something like `x.dtypes.map(size_of).sum() * len(x)` where `size_of` would take the `dtype` and compute it's size or return one from cache.

Reply:
Mapping over all the columns with `x.dtypes.map` will still be quite slow with a dataframe with lots of columns. Also invalidating cache may be hard?

Reply:
It should be rather fast. You cloud do something like:

```python
sum([
    size_of(dtype) * count
    for dtype, count in x.dtypes.value_counts().items()
]) * len(x)
```


The harder part is implementing `size_of`

Reply:
Would you be fine if, in this PR, I just add a watchdog that does:

```python
# It seems a big number of rows does not impact performance, only columns
if len(x.columns) < 10_000:
    return x.memory_usage().sum()
else:
    return "?"
```

And we can open a follow-up issue for a faster calculation of the memory usage, pointing to this discussion? 

This would at least fix the crashing issue we're seeing on our side.

Reply:
Sure, souds fine!